src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -21,6 +27,12 @@ public static final String reverseWithStringConcat(String string) {
     }
 
     public static final String reverseWithStringBuilder(String string) {
+        if (string == null) {
```

Comment:
Check for string null to confirm string exists, returning and empty string if it is null.